mars-cli/mars_lib/isa_json.py

```diff
@@ -420,3 +422,56 @@ def update_isa_json(isa_json: IsaJson, repo_response: RepositoryResponse) -> Isa
 
     isa_json.investigation = investigation
     return isa_json
+
+
+def map_data_files_to_repositories(
+    files: List[str], isa_json: IsaJson
+) -> Dict[str, List[str]]:
+    # Note: This works well in
+    df_map: Dict[str, List[str]] = {}
+    assays: List[Assay] = [
+        assay for study in isa_json.investigation.studies for assay in study.assays
+    ]
+
+    files_dicts = [{"full_name": f, "short_name": f.split("/")[-1]} for f in files]
+    remaining_files = files_dicts.copy()
+    for assay in assays:
+        target_repo_comment: Comment = detect_target_repo_comment(assay.comments)
+        # This is an effect of everything being optional in the Comment model.
+        # Should we decide to make the value mandatory, this guard clause would not be necessary anymore.
+        if target_repo_comment.value is None:
+            raise ValueError(
+                f"At least one assay in the ISA-JSON has no '{TARGET_REPO_KEY}' comment. Mapping not possible. Make sure all assays in the ISA-JSON have this comment!"
+            )
+        assay_data_files = [df.name for df in assay.dataFiles]
+
+        # Check if the files in the ISA-JSON are present in the command
+        # If not, raise an error
+        for adf in assay_data_files:
+            if adf not in [fd["short_name"] for fd in files_dicts]:
+                raise ValueError(
+                    f"""Assay for repository '{target_repo_comment.value}' has encountered a mismatch while mapping the data files to the ISA-JSON.
+                Data File '{adf}' is missing in the data files passed in the command:
+                {files}
+                Please correct the mismatch!"""
+                )
+            else:
+                remaining_files.remove(
+                    next(fd for fd in files_dicts if fd["short_name"] == adf)
+                )
```

Comment:
This removes the first occurrence, correct? It could be confusing if the user accidentally repeats a filename in the command and then still sees the warning that it will be skipped.  Could do something like
```
remaining_files = [fd for fd in remaining_files if fd["short_name"] == adf]
```
to get them all.

Reply:
Good catch! But should we allow the user to pass a file multiple times? Or are there situations where a data file is submitted to multiple repositories? (Honest question...)

Also, shouldn't it be this then:
```python
remaining_files = [fd for fd in remaining_files if fd["short_name"] != adf]
```

Reply:
Yes of course, your `remaining_files` filter is the right one :+1:

For submitting the same data files to multiple repositories, this is indeed a good question... I feel like this shouldn't happen but maybe it's something we'll have to revisit in the future.